### PR TITLE
Disable reverse proxy by default

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: spec/migration_spec.rb [hunk 1]
```diff
@@ -141,3 +141,21 @@
...
+    context 'reset reverse_proxy to default value to disable by default' do
+      let(:answers) do
+        {
+          'foreman_proxy_content' => {
+            'reverse_proxy' => true,
+          },
+        }
+      end
+
+      it 'changes the default to Redis' do
```
Comment (ekohl) [inline spec/migration_spec.rb]: Copy-paste error?

File: spec/migration_spec.rb [hunk 2]
```diff
@@ -141,3 +141,21 @@
     end
   end
 end
+
+migration '240814115647-disable-reverse-proxy-by-default' do
```
Comment (ekohl) [inline spec/migration_spec.rb]: I swear I created a PR for it, but I only had a local branch. Now I opened https://github.com/theforeman/foreman-installer/pull/959 to split it out over multiple files. I'd prefer that over one huge file.

**Review by ekohl — CHANGES_REQUESTED** [review]
:+1: for the migration itself, just some comments on the tests.

File: spec/migration_spec.rb (hunk 1 above)
Comment (ehelms) [reply spec/migration_spec.rb]: :facepalm:

File: spec/migration_spec.rb (hunk 2 above)
Comment (ehelms) [reply spec/migration_spec.rb]: Rebased on your merged split out

File: spec/migrations/240814115647-disable_reverse_proxy_by_default_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    context 'reset reverse_proxy to default value to disable by default' do
+      let(:answers) do
+        {
+          'foreman_proxy_content' => {
+            'reverse_proxy' => true,
+          },
+        }
+      end
+
+      it 'changes the default to the puppet module' do
```
Comment (ekohl) [inline spec/migrations/240814115647-disable_reverse_proxy_by_default_spec.rb]: `it` describes the expected behavior. How about:
```suggestion
      it 'removes the stored answer' do
```

File: spec/migrations/240814115647-disable_reverse_proxy_by_default_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
+
+migration '240814115647-disable-reverse-proxy-by-default' do
+  scenarios %w[foreman-proxy-content] do
+    context 'reset reverse_proxy to default value to disable by default' do
```
Comment (ekohl) [inline spec/migrations/240814115647-disable_reverse_proxy_by_default_spec.rb]: `context` is used to describe the context, typically the `let` blocks. How about
```suggestion
    context 'with a stored answer' do
```

Not asking for it here, but from that you can also easily see the context `without a stored answer`. That could have an `it "doesn't crash" do` or something like that.

At least, that's how I interpret them and structure my RSpec tests. See https://rspec.help/getting-to-know-describe-context/ or https://rspec.help/rspec/basics/ if you're interested in more.

**Review by ekohl — APPROVED** [review]
Sorry for not suggesting them earlier. It looks ok, but we can improve the descriptions a bit.

**Pull request merged**